Make sure we set APIServerCount

justinsb opened the pull request:
It looks like it got lost in a refactor.  Add a unit test, and move
initialization to code (and have the code self-check as well).

Also we can now have a fairly long code comment about the reasons why
this is such a mess...

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.kubernetes.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.kubernetes.io/reviews/kubernetes/kops/1048)
<!-- Reviewable:end -->

chrislovecnm commented:
Review status: 0 of 5 files reviewed at latest revision, 2 unresolved discussions, some commit checks failed.

---

*[upup/models/config/components/kube-apiserver/kube-apiserver.options, line 19 at r1](https://reviewable.kubernetes.io:443/reviews/kubernetes/kops/1048#-KY7A0Zo2ckbysntyftK:-KY7A0Zo2ckbysntyftL:b-xdvf8t) ([raw file](https://github.com/kubernetes/kops/blob/5a4cdceb56cd86c3e6b51715d312c28ba5b92144/upup/models/config/components/kube-apiserver/kube-apiserver.options#L19)):*
> ```
>   AllowPrivileged: true
>   Image: {{ Image "kube-apiserver" }}
>   APIServerCount: {{ APIServerCount }}
> ```

Do you need a EOF here?

---

*[upup/pkg/fi/cloudup/populatecluster_test.go, line 335 at r1](https://reviewable.kubernetes.io:443/reviews/kubernetes/kops/1048#-KY7A7hft-krHyMyaq2V:-KY7A7hft-krHyMyaq2W:b32nuj5) ([raw file](https://github.com/kubernetes/kops/blob/9c99d3f57d1744c81bc9f444699da1761c2cd1f0/upup/pkg/fi/cloudup/populatecluster_test.go#L335)):*
> ```Go
> 		t.Fatalf("Unexpected APIServerCount: %v", fi.IntValue(full.Spec.KubeAPIServer.APIServerCount))
> 	}
> }
> ```

WAHOO .. unit tests :)

---


*Comments from [Reviewable](https://reviewable.kubernetes.io:443/reviews/kubernetes/kops/1048)*
<!-- Sent from Reviewable.io -->

chrislovecnm commented:
Review status: 0 of 5 files reviewed at latest revision, 3 unresolved discussions, some commit checks failed.

---

*[pkg/model/components/apiserver.go, line 32 at r1](https://reviewable.kubernetes.io:443/reviews/kubernetes/kops/1048#-KY7ARj2GmchXYROv-dM:-KY7ARj2GmchXYROv-dN:b-6gx7n4) ([raw file](https://github.com/kubernetes/kops/blob/9c99d3f57d1744c81bc9f444699da1761c2cd1f0/pkg/model/components/apiserver.go#L32)):*
> ```Go
> var _ loader.OptionsBuilder = &KubeAPIServerOptionsBuilder{}
> 
> func (b *KubeAPIServerOptionsBuilder) BuildOptions(o interface{}) error {
> ```

Can we get a code comment, so that we can lint these files eventually?

---


*Comments from [Reviewable](https://reviewable.kubernetes.io:443/reviews/kubernetes/kops/1048)*
<!-- Sent from Reviewable.io -->

chrislovecnm commented:
Reviewed 5 of 5 files at r1.
Review status: all files reviewed at latest revision, 3 unresolved discussions, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.kubernetes.io:443/reviews/kubernetes/kops/1048)*
<!-- Sent from Reviewable.io -->

justinsb commented:
Let's stop using reviewable: https://github.com/kubernetes/kubernetes/issues/38087

EOL: added.

Added a linter comment for the class.  The method is implementing an interface, so I'm loathe to comment it...
